lib/ansible/module_utils/basic.py

```diff
@@ -1832,9 +1836,24 @@ def safe_eval(self, value, locals=None, include_exceptions=False):
     def _check_type_str(self, value):
         if isinstance(value, string_types):
             return value
-        # Note: This could throw a unicode error if value's __str__() method
-        # returns non-ascii.  Have to port utils.to_bytes() if that happens
-        return str(value)
+
+        # Note: str() could throw a unicode error if value's __str__() method
+        # returns non-ascii. Have to port utils.to_bytes() if that happens.
+        #
+        # Ignore, warn, or error when converting to a string.
+        # The current default is to warn. Change this in Anisble 2.12 to error.
+        common_msg = 'The value {0!r} (type {0.__class__.__name__}) in a string field was converted to {1!r} (type string). {2}'
+        if self._string_conversion_action == 'error':
+            extra_msg = 'Quote the entire value to ensure it does not change.'
+            msg = common_msg.format(value, str(value), extra_msg)
+            self.fail_json(msg=msg)
+        elif self._string_conversion_action == 'ignore':
+            return str(value)
+        else:
+            extra_msg = 'If this does not look like what you expect, quote the entire value to ensure it does not change.'
+            msg = common_msg.format(value, str(value), extra_msg)
+            self.warn(msg)
+            return str(value)
```

Comment:
should it be to_text instead of str? i know this is issue with original code also

Reply:
From looking at the code at the time that comment was made, we didn't have `module_utils._to_text()` at the time. I'm going to infer that's the reason we used `str()` here. I'll change this.